lib/DeleteItemBuilder.js

```diff
@@ -13,13 +13,15 @@ function DeleteItemBuilder(options) {
 require('util').inherits(DeleteItemBuilder, Builder)
 
 DeleteItemBuilder.prototype.execute = function () {
-  var queryData = new DynamoRequest(this.getOptions())
+  var req = new DynamoRequest(this.getOptions())
     .setTable(this._tablePrefix, this._table)
     .returnConsumedCapacity()
     .setHashKey(this._hashKey, true)
-    .setRangeKey(this._rangeKey, true)
     .setExpected(this._conditions)
-    .build()
+    
+  if (this._rangeKey) req.setRangeKey(this._rangeKey, true)
```

Comment:
It could make sense. Changing DynamoRequest requires much less code. The approach in this PR is more consistent with the public Builder API: don't have a range key? Then don't call setRangeKey. I like that both PK setters don't silently ignore invalid params, but I'm not sure there's much benefit in doing so in this case—this isn't a public API people use.

Another option would be to add a newRequest function (or something with some better name) to the Builder, something like this:
```js
Builder.prototype.newRequest = function newRequest() {
  var req = new DynamoRequest(this.getOptions());

  req
    .setTable(this._tablePrefix, this._table)
    .returnConsumedCapacity()
    .setHashKey(this._hashKey, true)
    .setExpected(this._conditions)

  if (this._rangeKey) req.setRangeKey(this._rangeKey, true);

  return req;
};
```

That way, at least all the different Builders don't need to remember to check the range key before calling setRangeKey.